### ISSUE TITLE
Source command failed to work when file name has a space

### ISSUE THREAD

**Issue by tomassabat** [graknlabs/console#139]
1. Run `grakn-core-2.0.0-alpha-9`
2. Run grakn console
3. Create a database
4. Open transaction
5. Run the `source` command with a file that has a space in the file path, for example: `source /Users/tasabat/Desktop/Financial-Knowledge-Graph-2.0\ .gql`
Error shown: 
```
[GQL03] Graql Error: There is a syntax error at line 1:
source /Users/tasabat/Desktop/Financial-Knowledge-Graph-2.0 .gql 
^
no viable alternative at input 'source'
[GQL03] Graql Error: There is a syntax error at line 1:
source /Users/tasabat/Desktop/Financial-Knowledge-Graph-2.0 .gql 
^
mismatched input 'source' expecting {'match', 'define', 'undefine', 'insert', 'compute'}
```
If we remove the space in the file path, then the loading works.

**Comment by haikalpribadi** [comment]
@tomassabat this is a common bug in any terminal application. Have you tried escaping the space character with backslash?

@lriuui0x0 do you know if we can quote the source filename to not need escaping any characters?

**Comment by tomassabat** [comment]
@haikalpribadi Yes I believe so, see the escape before the space at the end of the filepath: `...2.0\ .gql`. This is added automatically when I drag and drop a file into the terminal. Putting the file path in quotes also throws the same error, like so: 
`source "/Users/tasabat/Desktop/Financial-Knowledge-Graph-2.0 .gql"`

However, in 1.8 the file path (with the same escape) does work: 
`./grakn console -k 'test' -f /Users/tasabat/Desktop/Financial-Knowledge-Graph-2.0\ .gql`

**Comment by lriuui0x0** [comment]
@haikalpribadi @tomassabat The current command line argument processing is very rudimentary, which only deals with arguments without spaces in it. If we want to implement the quoting or escape characters, we need to implement a parser of the command line arguments.

**Comment by haikalpribadi** [comment]
Okay then let's leave this in backlog for now, @lriuui0x0 - but can we provide a better error message? Also I'm confused: why does the error message Graql error? and why does it look like it uses ANTLR?

@tomassabat for now just please make sure to not have spaces in file names for now.

**Comment by tomassabat** [comment]
Will do thanks @haikalpribadi

**Comment by JRWest2000** [comment]
Can I suggest a smaller fix catch the error and throw something meaningful I wasted way more time trying to figure out what was happing then a simple space in a file deserved. In fact since this was lets give it a try and see if it looks like it is worth more time to dive deeper into it's functionality and I am only giving myself 2 hours to play before I move on, test drive. I almost walked away.

**Comment by flyingsilverfin** [comment]
I'm cleaning this up in https://github.com/graknlabs/console/pull/150 @JRWest2000  @tomassabat , at least the error message is then obvious!

**Comment by lriuui0x0** [comment]
@haikalpribadi Sorry for the late reply, the reason used to be we check for a command by the first token and the number of arguments. So in this case the first token is `source` but it has three tokens in total, which is different from two tokens it expects, so it thought it was not a source command but fallback to graql query.

This behaviour seems have already been improved by @flyingsilverfin

**Comment by haikalpribadi** [comment]
Ah, I see. That logic is flawed, @lriuui0x0 . Once you know the first token, you should route it a different decision tree. Every initial tokens are unique commands, so you don't need to fall back to a different logic once they are identified. Further parsing of the number of arguments following that token should be done independently, which would allow you to provide better error message.
